guides/source/wishlists.md

```diff
@@ -0,0 +1,1545 @@
+
+<%= button_to "Remove", store_subscriber_path(@subscriber), method: :delete, data: { turbo_confirm: "Are you sure?" } %>
+```
+
+Finally, add the link to the sidebar layout:
+
+```erb#14
+<%= content_for :content do %>
+  <section class="settings">
+    <nav>
+      <h4>Account Settings</h4>
+      <%= link_to "Profile", settings_profile_path %>
+      <%= link_to "Email", settings_email_path %>
+      <%= link_to "Password", settings_password_path %>
+      <%= link_to "Account", settings_user_path %>
+
+      <% if Current.user.admin? %>
+        <h4>Store Settings</h4>
+        <%= link_to "Products", store_products_path %>
+        <%= link_to "Users", store_users_path %>
+        <%= link_to "Subscribers", store_subscribers_path %>
+        <%= link_to "Wishlists", store_products_path %>
+      <% end %>
+    </nav>
+
+    <div>
+      <%= yield %>
+    </div>
+  </section>
+<% end %>
+
+<%= render template: "layouts/application" %>
+```
+
+Now you can view, filter, and remove subscribers in the store's admin area. Try
+it out!
+
+## Adding Links To Products
+
+Now that we've added filters, we can add links to the Product show page for
+viewing wishlists and subscribers for a specific product.
+
+```erb#18-21
+<p><%= link_to "Back", store_products_path %></p>
+
+<section class="product">
+  <%= image_tag @product.featured_image if @product.featured_image.attached? %>
+
+  <section class="product-info">
+    <% cache @product do %>
+      <h1><%= @product.name %></h1>
+      <%= @product.description %>
+    <% end %>
+
+    <%= link_to "View in Storefront", @product %>
+    <%= link_to "Edit", edit_store_product_path(@product) %>
+    <%= button_to "Delete", [ :store, @product ], method: :delete, data: { turbo_confirm: "Are you sure?" } %>
+  </section>
+</section>
+
+<section>
+  <%= link_to pluralize(@product.wishlists_count, "wishlist"), store_wishlists_path(product_id: @product) %>
+  <%= link_to pluralize(@product.subscribers.count, "subscriber"), store_subscribers_path(product_id: @product) %>
+</section>
+```
+
+## Testing Wishlists
+
+Let's write some tests for the functionality we just built.
+
+### Adding Fixtures
+
+First, we need to update the fixtures in `test/fixtures/wishlist_products.yml`
+so they refer to the product fixtures we have defined:
+
+```yaml
+one:
+  product: tshirt
+  wishlist: one
+
+two:
+  product: tshirt
+  wishlist: two
+```
+
+Let's also add another product fixture in `test/fixtures/products.yml` to test
+with:
+
+```yaml#5-7
+tshirt:
+  name: T-Shirt
+  inventory_count: 15
+
+shoes:
+  name: shoes
+  inventory_count: 0
+```
+
+### Testing `to_param`
+
+Our Wishlist model's friendly URLs are an important feature, so let's start by
+writing a test for it.
+
+Open `test/models/wishlist_test.rb` and add this test:
+
+```ruby
+require "test_helper"
+
+class WishlistTest < ActiveSupport::TestCase
+  test "has friendly urls" do
```

Comment:
IDK how I feel about this test. This seems mostly to be testing Rails code (which we should assume is tested). Maybe we could test that we can find it? Or maybe no test for this at all, since the integration specs will catch it.

Reply:
Agreed, I'll swap this with a test for `filter_by` since that is something actually worth unit testing.